### DISCUSSION TITLE
There is a problem with the object value Deep watching

### DISCUSSION THREAD

**Discussion by KRHugo** [alpinejs/alpine#4076] [7. Bugs (archived)]
```
<div x-data="{ foo: { bar: 'baz' }}" x-init="$watch('foo', (value, oldValue) => console.log(value, oldValue))">
    <button @click="foo.bar = 'bob'">Update</button>
</div>
```

https://alpinejs.dev/magics/watch
In the example page, the actual results are different from those expected by Deep watching.

Expected result: `new - {bar: 'bob'}, old - {bar: 'baz'}`
Actual Results: `new - {bar: 'bob'}, old - {bar: 'bob'}`

alpine version 3.13.5

**Comment by ekwoka** [comment]
$watch internally does not store a copy of the old value.

In this case the old value of foo is also the same object that is the new value of foo.

That's how objects in JavaScript work.
